Add KeyHandler to RateLimitConfig

ugoArangino opened the pull request:
This adds a `KeyHandler` to `RateLimitConfig`

The `KeyHandler` takes a `RouterRequest ` and returns a `String` for the `keyStore`.
This extends Kitura-RateLimit so that it is not bound to the *IP address* and can use e.g. the *ClientID* from a *REST* application.

ugoArangino committed:
Add KeyHandler to RateLimitConfig

attheodo commented:
@ugoArangino cool stuff, thanks for you contribution. Care to update the `README` and document the `keyHandler` property and usage? Also make sure you also update the CHANGELOG at the bottom. I'd be more than happy to merge then! ⚡️

ugoArangino committed:
Move KeyHandler into a separate file with docu

ugoArangino committed:
Add Example

* To the Example project
* To the README

ugoArangino commented:
@attheodo I have updated the Example in the project and the `README` and moved the `KeyHandler` into a operate file with docu.

I think the CHANGELOG should be updated when you have merged my RP and tagged it.

attheodo commented:
@ugoArangino sorry, what happened here? I may have lost track of that one and I apologize. Should we re-open and merge?
